Automattic/jetpack
# Limit *_comment_meta actions that are synced

Update the  hook of Sync's Comment module to not send meta actions when the comment_type is not whitelisted.

Tests added for known and unknown type/meta key use cases.

#### Changes proposed in this Pull Request:
* Updates the Comment Sync module to utilize the comment_type_whitelist when enqueuing comment_meta actions

Actions Impacted
* added_comment_meta
* updated_comment_meta
* deleted_comment_meta

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Updates existing functionality to utilize whitelists

#### Testing instructions:

1. Ensure Sync is enabled on your test site
1. Create a comment with a non-whitelisted comment_type ("excalibur", "test", etc)
1. Add comment_meta with a whitelisted meta key `hc_avatar`
1. Verify no `added_comment_meta` action is queued
1. Add comment_meta with a non whitelisted meta key 'gobbler'
1. Verify no `added_comment_meta` action is queued

#### Proposed changelog entry for your changes:
* N/A - whitelist was existing functionality but not applied to all meta actions.
